File: common-parent/common-utils/src/main/java/love/forte/simbot/common/utils/convert/ConverterManagerBuilder.java
package love.forte.simbot.common.utils.convert;

import java.lang.reflect.Type;

/**
 *
 * {@link ConverterManager} 的构建器。
 *
 * @author <a href="https://github.com/ForteScarlet"> ForteScarlet </a>
 */
public interface ConverterManagerBuilder {

    /**
     * 注册一个转化器
     * @param target 目标类型
     * @param converter 转化器
     * @return 当前builder
     */
    ConverterManagerBuilder register(Type target, Converter<?> converter);

    /**
     * 构建一个 {@link ConverterManager} 实例。
     * @return {@link ConverterManager} 实例
     */
    ConverterManager build();
}

File: common-parent/common-utils/src/main/java/love/forte/simbot/common/utils/convert/ConverterManagerBuilder.kt
package love.forte.simbot.common.utils.convert


/**
 *
 * [ConverterManager]的构建器
 *
 * @author ForteScarlet -> https://github.com/ForteScarlet
 */
interface ConverterManagerBuilder2 {

    /**
     * 注册一个转化器。
     * @param cover 如果对应的转化已经存在，是否覆盖。
     */
    fun <T> register(target: Class<T>, cover: Boolean, converter: Converter<T>): Boolean


    /**
     * 构建要一个 [ConverterManager]
     */
    fun build(): ConverterManager
}


